### PR TITLE
Release notes v21.2.0 beta.1 and v21.2.0-beta.2

### LINKED ISSUE #10715: v21.2.0-beta.1 Release Notes
Prep date: tbd
Release date: Sep 20, 2021
SHA candidate: tbd
Release Manager: tbd
Issue: tbd
[Docs To Do](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-16.Updatedocs):

- [ ] Run script to generate release notes
- [ ] Update docs site
- [ ] Edit release notes - Additions/changes/omissions to the release notes should be left in this issue's comments
- [ ] Once PR is merged / release notes are live, notify @meagan-g in the #releases Slack channel
- [ ] Email releases@ with link to published release notes
- [ ] Generate release notes issues (cc @jseldess)

### PR DESCRIPTION
**Pull request by kathancox**
Closes #10715, #10716

Release notes for v21.2.0-beta.1 and beta.2

For review: Note that this includes **already edited** notes from the alpha that were never released. Markdown comments show the items that do not need review

### CODE REVIEW

**Review by ericharmeling — COMMENTED** [review]
Looks good! Just a couple nits and a few suggestions.

> Reviewed 2 of 4 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/11722#-:-Mk88bIW6DwHCYI4Wozy:b-hjggo5)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @kathancox)

___
*[releases/v21.2.0-beta.1.md, line 4 at r2](https://reviewable.io/reviews/cockroachdb/docs/11722#-Mk7VHEh7LjQYOTRGfHt:-Mk7VHEh7LjQYOTRGfHu:b8vx8qm) ([raw file](https://github.com/cockroachdb/docs/blob/f2cc9fe1664a3044bec3fd3e577cb4b84bcfbffe/releases/v21.2.0-beta.1.md#L4)):*
nit:

> ```
>  since version 21.2.0-alpha.1
> ```

Should probably remove this bit (even though this sentence just part of the metadata for the page).
___
*[releases/v21.2.0-beta.1.md, line 55 at r2](https://reviewable.io/reviews/cockroachdb/docs/11722#-Mk7bEbAAw8XgAJnIjaL:-Mk7bEbAAw8XgAJnIjaM:bym657k) ([raw file](https://github.com/cockroachdb/docs/blob/f2cc9fe1664a3044bec3fd3e577cb4b84bcfbffe/releases/v21.2.0-beta.1.md#L55)):*
suggestion:

> ```
> Extended the HTTP server on the sqlproxy to support `pprof` [debug endpoints](../v21.2/monitoring-and-alerting.html#raw-status-endpoints) to help users troubleshoot the sqlproxy. [#69594][#69594]
> ```

I don't think "sqlproxy" is exposed to users. I think it's a backend for the HTTP endpoints we expose (that's kind of what it looks like [here](https://github.com/alyshanjahani-crl/cockroach/blob/master/pkg/ccl/sqlproxyccl/server.go#L32), anyways).

Perhaps this might be a little more accurate (but less informative?):

```
CockroachDB now supports new [debug endpoints](../v21.2/monitoring-and-alerting.html#raw-status-endpoints) to help users with troubleshooting. [#69594][#69594]
```
___
*[releases/v21.2.0-beta.1.md, line 114 at r2](https://reviewable.io/reviews/cockroachdb/docs/11722#-Mk7_6BH4xHCxjrUeZrP:-Mk7_6BH4xHCxjrUeZrQ:b-bstbyz) ([raw file](https://github.com/cockroachdb/docs/blob/f2cc9fe1664a3044bec3fd3e577cb4b84bcfbffe/releases/v21.2.0-beta.1.md#L114)):*
suggestion:

> ```
> 1,000,000 rows.
> ```

```
`1000000` (1,000,000 rows).
```
___
*[releases/v21.2.0-beta.1.md, line 293 at r2](https://reviewable.io/reviews/cockroachdb/docs/11722#-Mk7_g1X5bLUoORJUqot:-Mk7_g1X5bLUoORJUqou:b-t9jbp8) ([raw file](https://github.com/cockroachdb/docs/blob/f2cc9fe1664a3044bec3fd3e577cb4b84bcfbffe/releases/v21.2.0-beta.1.md#L293)):*
suggestion:

> ```
> Operators will potentially have access to a CSV file containing active queries from the last run of the queryprofile upon OOM.
> ```

```
On receiving out-of-memory errors from CockroachDB, operators now have access to a CSV file containing the last active queries.
```

(TBH, IDK what the queryprofile is. We don't document it anywhere, so it might be best to leave it out of the note)
___
*[releases/v21.2.0-beta.1.md, line 409 at r2](https://reviewable.io/reviews/cockroachdb/docs/11722#-Mk84nhhEbw_Ugtc5Yqg:-Mk84nhhEbw_Ugtc5Yqh:bwjoggc) ([raw file](https://github.com/cockroachdb/docs/blob/f2cc9fe1664a3044bec3fd3e577cb4b84bcfbffe/releases/v21.2.0-beta.1.md#L409)):*
nit:

> ```
> Database page
> ```

```
Database Page
```
___
*[releases/v21.2.0-beta.1.md, line 469 at r2](https://reviewable.io/reviews/cockroachdb/docs/11722#-Mk84ix91CPHW3Qfd5IM:-Mk84ix91CPHW3Qfd5IN:b-hresw2) ([raw file](https://github.com/cockroachdb/docs/blob/f2cc9fe1664a3044bec3fd3e577cb4b84bcfbffe/releases/v21.2.0-beta.1.md#L469)):*
nit:

> ```
> ALTER PRIMARY KEY
> ```

```
`ALTER PRIMARY KEY`
```

**Review by ericharmeling — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/11722#-:-Mk8dMd14LX0bPKAFit0:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @kathancox)

File: _config_base.yml [hunk 1]
```diff
@@ -84,10 +84,10 @@ release_info:
     start_time: 2021-8-09 11:01:26.34274101 +0000 UTC
     version: v21.1.7
   v21.2:
-    build_time: 2021/08/12 11:00:26 (go1.13.4)
+    build_time: 2021/09/22 11:00:26 (go1.13.4)
     docker_image: cockroachdb/cockroach-unstable
     name: v21.2
-    start_time: 2021-08-12 11:01:26.34274101 +0000 UTC
-    version: v21.2.0-alpha.1
+    start_time: 2021-09-22 11:01:26.34274101 +0000 UTC
```
Comment (jseldess) [inline _config_base.yml:90]: We'll need to update the date to match when we actually release.

File: _data/releases.yml [hunk 2]
```diff
@@ -268,6 +268,8 @@
       no_windows: true
 - title: Testing releases
   releases:
+    - date: Sep 22, 2021
```
Comment (jseldess) [inline _data/releases.yml:271]: Same.

File: releases/v21.2.0-beta.1.md [hunk 3]
```diff
@@ -0,0 +1,1346 @@
+---
+title: What&#39;s New in v21.2.0-beta.1
+toc: true
+summary: Additions and changes to CockroachDB version 21.2.0-beta.1.
+---
+
+## September 21, 2021
```
Comment (jseldess) [inline releases/v21.2.0-beta.1.md:7]: Same.

File: releases/v21.2.0-beta.1.md [hunk 4]
```diff
@@ -0,0 +1,1346 @@
...
+[#69469]: https://github.com/cockroachdb/cockroach/pull/69469
+[#69470]: https://github.com/cockroachdb/cockroach/pull/69470
+[#69478]: https://github.com/cockroachdb/cockroach/pull/69478
+[#69480]: https://github.com/cockroachdb/cockroach/pull/69480
+[#69481]: https://github.com/cockroachdb/cockroach/pull/69481
+[#69483]: https://github.com/cockroachdb/cockroach/pull/69483
+[#69491]: https://github.com/cockroachdb/cockroach/pull/69491
+[#69502]: https://github.com/cockroachdb/cockroach/pull/69502
+[#69577]: https://github.com/cockroachdb/cockroach/pull/69577
+[0056828fd]: https://github.com/cockroachdb/cockroach/commit/0056828fd
```
Comment (jseldess) [inline releases/v21.2.0-beta.1.md:1218]: We can remove this line and all lines following. They are left-over links to commits that aren't linked to from this page.

**Review by jseldess — APPROVED** [review]
LGTM, pending date updates.

**Comment by kathancox** [comment]
___
*[releases/v21.2.0-beta.2.md, line 41 at r5](https://reviewable.io/reviews/cockroachdb/docs/11722#-Mkccp1V75CgrMwB7K_C:-Mkccp1V75CgrMwB7K_D:byhln86) ([raw file](https://github.com/cockroachdb/docs/blob/c2b5c1163af99551f42964b134128913dc88999f/releases/v21.2.0-beta.2.md#L41)):*
> ```Markdown
> - The query logging enabled by `sql.telemetry.query_sampling.enabled` now avoids considering SQL statements issued internally by CockroachDB itself. [#70358][#70358]
> - [`IMPORT INTO`](../v21.2/import-into.html) now supports UDT for default and computed columns. [#70270][#70270]
> - [`IMPORT INTO`](../v21.2/import-into.html) regional by row tables is supported. [#70270][#70270]
> ```

Maybe could link to multiregion-overview#regional-by-row-tables on "regional by row" and add in "now" or switch the tense of the sentence to match other sentences. Perhaps "[`IMPORT INTO`](../v21.2/import-into.html) a [regional by row](multiregion-overview#regional-by-row-tables] table is now supported."

**Comment by kathancox** [comment]
___
*[releases/v21.2.0-beta.2.md, line 45 at r5](https://reviewable.io/reviews/cockroachdb/docs/11722#-MkcdtAPDY10977kroqJ:-MkcdtAPDY10977kroqK:bz4yaxs) ([raw file](https://github.com/cockroachdb/docs/blob/c2b5c1163af99551f42964b134128913dc88999f/releases/v21.2.0-beta.2.md#L45)):*
> ```Markdown
> ### Operational changes
> 
> - The meaning of the recently introduced `transaction_rows_written_err` and `transaction_rows_read_err` (as well as the corresponding `_log` variables) have been adjusted a bit to indicate the largest number of rows that is still allowed. In other words, originally reaching the limit would result in an error, and now only exceeding the limit would. [#70014][#70014]
> ```

Nit: not sure if "a bit" is necessary here.

**Comment by kathancox** [comment]
___
*[releases/v21.2.0-beta.2.md, line 63 at r5](https://reviewable.io/reviews/cockroachdb/docs/11722#-MkceGKND3qWA4cvw_eZ:-MkceGKND3qWA4cvw_e_:b-8s6u3u) ([raw file](https://github.com/cockroachdb/docs/blob/c2b5c1163af99551f42964b134128913dc88999f/releases/v21.2.0-beta.2.md#L63)):*
> ```Markdown
>     ~~~
> 
>     This captures all messages on all channels except the `STORAGE` channel, plus the messages at severity `WARNING` or higher from `STORAGE`.  Note: the previous syntax remains supported. When `channel` is specified without explicit severities, the `filter` attribute is used as default (like previously). [#70411][#70411]
> ```

Nit: maybe ....as "the" default...

**Comment by kathancox** [comment]
___
*[releases/v21.2.0-beta.2.md, line 77 at r5](https://reviewable.io/reviews/cockroachdb/docs/11722#-MkcelR_E3zFa-d_zR0b:-MkcelR_E3zFa-d_zR0c:b-fzrs9h) ([raw file](https://github.com/cockroachdb/docs/blob/c2b5c1163af99551f42964b134128913dc88999f/releases/v21.2.0-beta.2.md#L77)):*
> ```Markdown
> ### Bug fixes
> 
> - Columns that were hidden by default where not being displayed when selected. This commits fixes the behavior. [#70054][#70054]
> ```

Typo: s/where/were... "...by default were not being displayed..."
At the end, I think there is a typo on commit, maybe: "This commit fixes this behavior"

**Comment by kathancox** [comment]
___
*[releases/v21.2.0-beta.2.md, line 82 at r5](https://reviewable.io/reviews/cockroachdb/docs/11722#-MkcfYll1V_STeGfT1gk:-MkcfYll1V_STeGfT1gl:bk69slo) ([raw file](https://github.com/cockroachdb/docs/blob/c2b5c1163af99551f42964b134128913dc88999f/releases/v21.2.0-beta.2.md#L82)):*
> ```Markdown
> - DNS unavailability during range 1 leaseholder loss will no longer cause significant latency increases for queries and other operations. [#70135][#70135]
> - Last Execution Timestamp is now properly updating. [#70297][#70297]
> - Fixed a bug in full cluster restores where dropped descriptor revisions would cause the restore to fail. [#70368][#70368]
> ```

Could link to [restore](restore.html) here?

**Comment by kathancox** [comment]
___
*[releases/v21.2.0-beta.2.md, line 83 at r5](https://reviewable.io/reviews/cockroachdb/docs/11722#-MkcfrVrAgHrRngEWNYe:-MkcfrVrAgHrRngEWNYf:b-afa6bg) ([raw file](https://github.com/cockroachdb/docs/blob/c2b5c1163af99551f42964b134128913dc88999f/releases/v21.2.0-beta.2.md#L83)):*
> ```Markdown
> - Last Execution Timestamp is now properly updating. [#70297][#70297]
> - Fixed a bug in full cluster restores where dropped descriptor revisions would cause the restore to fail. [#70368][#70368]
> - Default columns were displayed on the **Statements** page on the CockroachCloud console when the user never made any selection. This is now fixed. [#70206][#70206]
> ```

Not sure, but would it be useful to link to this page here: cockroachcloud/statements-page.html

**Review by lnhsingh — COMMENTED** [review]
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/11722#-:-MkcnHcP678xCv9kdj4z:b-814g8k)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @ericharmeling, @jseldess, @kathancox, and @lnhsingh)

___
*[releases/v21.2.0-beta.2.md, line 41 at r5](https://reviewable.io/reviews/cockroachdb/docs/11722#-Mkccp1V75CgrMwB7K_C:-Mkcj5Mp97AZ_49Zahvr:bb6un1j) ([raw file](https://github.com/cockroachdb/docs/blob/c2b5c1163af99551f42964b134128913dc88999f/releases/v21.2.0-beta.2.md#L41)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

Maybe could link to multiregion-overview#regional-by-row-tables on "regional by row" and add in "now" or switch the tense of the sentence to match other sentences. Perhaps "[`IMPORT INTO`](../v21.2/import-into.html) a [regional by row](multiregion-overview#regional-by-row-tables] table is now supported."
</blockquote></details>

Done
___
*[releases/v21.2.0-beta.2.md, line 45 at r5](https://reviewable.io/reviews/cockroachdb/docs/11722#-MkcdtAPDY10977kroqJ:-MkcmncGE5Lqip2_TXUH:bb6un1j) ([raw file](https://github.com/cockroachdb/docs/blob/c2b5c1163af99551f42964b134128913dc88999f/releases/v21.2.0-beta.2.md#L45)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

Nit: not sure if "a bit" is necessary here.
</blockquote></details>

Done
___
*[releases/v21.2.0-beta.2.md, line 63 at r5](https://reviewable.io/reviews/cockroachdb/docs/11722#-MkceGKND3qWA4cvw_eZ:-MkcmrdH1vBhcF2PaR5u:bb6un1j) ([raw file](https://github.com/cockroachdb/docs/blob/c2b5c1163af99551f42964b134128913dc88999f/releases/v21.2.0-beta.2.md#L63)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

Nit: maybe ....as "the" default...
</blockquote></details>

Done
___
*[releases/v21.2.0-beta.2.md, line 77 at r5](https://reviewable.io/reviews/cockroachdb/docs/11722#-MkcelR_E3zFa-d_zR0b:-Mkcn0Hf5pG60rYppSzT:b3oh8n) ([raw file](https://github.com/cockroachdb/docs/blob/c2b5c1163af99551f42964b134128913dc88999f/releases/v21.2.0-beta.2.md#L77)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

Typo: s/where/were... "...by default were not being displayed..."
At the end, I think there is a typo on commit, maybe: "This commit fixes this behavior"
</blockquote></details>


Done.
___
*[releases/v21.2.0-beta.2.md, line 82 at r5](https://reviewable.io/reviews/cockroachdb/docs/11722#-MkcfYll1V_STeGfT1gk:-Mkcn6iF4KMlg5Or2juB:bb6un1j) ([raw file](https://github.com/cockroachdb/docs/blob/c2b5c1163af99551f42964b134128913dc88999f/releases/v21.2.0-beta.2.md#L82)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

Could link to [restore](restore.html) here?
</blockquote></details>

Done
___
*[releases/v21.2.0-beta.2.md, line 83 at r5](https://reviewable.io/reviews/cockroachdb/docs/11722#-MkcfrVrAgHrRngEWNYe:-MkcnEMiEfnEqY7lOsdt:bs7wumb) ([raw file](https://github.com/cockroachdb/docs/blob/c2b5c1163af99551f42964b134128913dc88999f/releases/v21.2.0-beta.2.md#L83)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

Not sure, but would it be useful to link to this page here: cockroachcloud/statements-page.html
</blockquote></details>

Done.

**Pull request merged**